Assets/Scripts/Interaction Scripts/ThirdPersonView.cs

```diff
@@ -0,0 +1,450 @@
+                        break;
+                    }
+
+                    if (!controllerInput.GetRightIndex())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        PlaceWaypoint(RightUI.transform.position);
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                    }
+                    else
+                    {
+                        /// TODO: continue line showing
+                    }
+                    break;
+
+                }
+
+                case ControllerState.MOVING_WAYPOINT:
+                {
+                    break;
+                }
+
+                case ControllerState.UNDOING:
+                {
+                    if (!controllerInput.GetRightA())
+                    {
+                        Undo();
+                        break;
+                    }
+                    if (controllerInput.GetRightMiddle())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                        /// TODO: make waypoint disappear
+                        break;
+                    }
+                    controllerState = ControllerState.IDLE;
+                    break;
+                }
+
+                case ControllerState.REDOING:
+                {
+                    if (!controllerInput.GetRightB())
+                    {
+                        Redo();
+                    }
+                    if (controllerInput.GetRightMiddle())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                        /// TODO: make waypoint disappear
+                        break;
+                    }
+                    controllerState = ControllerState.IDLE;
+                    break;
+                }
+
+            }
+
+        }
+
+
+        private void ScaleWorld()
+        {
+            // Get the scaling factor, and adjust its size.
+            float ScalingFactor = 1.0f + 0.2f * controllerInput.GetScalingFactor();
+            Vector3 ScalingVector = Vector3.Scale(World.transform.localScale, new Vector3(ScalingFactor, ScalingFactor, ScalingFactor));
+
+            //Checking Scaling Bounds
+            if (ScalingVector.sqrMagnitude > WorldScaleMin.sqrMagnitude && ScalingVector.sqrMagnitude < WorldScaleMax.sqrMagnitude)
+            {
+                // FIXME: Jank. and comments.
+                Vector3 A = World.transform.position;
+                Vector3 B = Pivot.transform.position;
```

Comment:
is pivot ever not at 0,0,0?

Reply:
I don't think so, but gives us the flexibility yo move it around. The code was inherited from ISAACS v1.